### ISSUE TITLE
Field-less enums: an issue potentially introduced with the version bump to 1.79

### ISSUE THREAD

**Issue by y1ca1** [verus-lang/verus#1339]
I'm pretty excited about this version bump as Rust 1.79 stabilized the inline `const` expressions. Thanks for the efforts!

 However, my existing codebase breaks at a field-less enum declaration:

```rust
 enum FieldLess {
    A = 0,
    B = 1,
    C = 2,
}
```

Verus reports:
```sh
error: constant evaluation of enum discriminant resulted in non-integer
```

A repro link to the playground: https://aurorar8-eb5c.andrew.cmu.edu/?version=stable&mode=basic&edition=2021&gist=e87455afc0f418ffddbf2d5b7a2bb149

 Hope this is fairly straightforward to resolve!

**Comment by tjhance** [comment]
probably the same root cause as: https://github.com/verus-lang/verus/issues/1334

**Comment by utaal** [comment]
@y1ca1: the repro link goes to somewhere with an expired certificate. Can you provide the repro so we can add it as a regression test? Thanks!

**Comment by y1ca1** [comment]
Yes, see [here](https://play.verus-lang.org/?version=stable&mode=basic&edition=2021&gist=58ae53f4db649d1ed18e971909b2d55e).

**Comment by utaal** [comment]
Thank you!
